p4-16/spec/P4-16-spec.mdk

```diff
@@ -3954,12 +4014,32 @@ typeArgumentList
     ;
 ~ End P4Grammar
 
-Function arguments are evaluated in order, left to right, before the
-function invocation takes place. The calling convention is
-copy-in/copy-out (Section [#sec-calling-convention]). For generic functions the type
-arguments can be explicitly specified in the function call. The compiler does not
-insert implicit casts for the arguments to methods or functions---the argument types
-must match the parameter types exactly.
+A function call or method invocation can optionally specify for each
+argument the corresponding parameter name.  It is illegal to use names
+only for some arguments: either all or no arguments should specify
+the parameter name.  Function arguments are evaluated in the order
+they appear, left to right, before the function invocation takes
+place.
+
+~ Begin P4Example
+extern void f(in bit<32> x, out bit<16> y);
+bit<32> xa = 0;
+bit<16> ya;
+f(xa, ya);  // match arguments by position
+f(x = xa, y = ya);  // match arguments by name
+f(y = ya, x = xa);  // match arguments by name in any order
```

Comment:
Is `ya` evaluated before `xa` in this case? The OrderArgument pass reorders the argument to the original order of x, y. As a result, backend only sees x, then y. Will the evaluation order be `xa`, then `ya` then?

Reply:
The  evaluation order is handled by the frontend. It only matters if evaluation has side-effects. The text indicates that evaluation is done in argument order and not parameter order. At the backend stage the argument expressions will never have side effects.